app/controllers/timeslots_controller.rb

```diff
@@ -0,0 +1,67 @@
+class TimeslotsController < ApplicationController
+  before_action :check_full_access, except: %i[index show]
+  before_action :set_timeslot, only: %i[show edit update destroy]
+
+  def index
```

Comment:
Move this logic to User model

Reply:
Service is not Model!!!

Reply:
@BohdanChaban  In order not to violate business logic, create a service to not overload the model.?